src/js/scheduler.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Schedule a callback. Supports both immediate and delayed callbacks.
+ * @param callback The callback to be scheduled
+ * @param timeout The delay in milliseconds before the callback is called
+ */
+export function scheduleCallback(callback: () => void, timeout: number = 0) {
+  if (timeout <= 0) {
+    const channel = new MessageChannel();
```

Comment:
I was also working on a faster loop for my own project and at first I was also creating a new MessageChannel for each call, but that ran into garbage collection issues on both Chrome and Firefox when I had too many concurrent coroutines. The garbage collector could not keep up cleaning up the channels and the page kept crashing after the memory usage reached ~20GB RAM.

This is my current implementation, but I still need to clean it up a bit: https://github.com/e-nikolov/mpyc-web/blob/master/mpyc-web-py/lib/weblooperV10.py

It uses a single channel to avoid the memory issues.

Reply:
I see. Thanks for letting me know and sharing your implementation!

When I did some simple checks in the browser devtools, it seemed like the MessageChannel and MessagePorts are garbage collected well. But I'll keep investigating.